nes-plugins/Sources/GeneratorSource/Generator.cpp

```diff
@@ -33,26 +34,58 @@ namespace NES
 
 void Generator::generateTuple(std::ostream& ostream)
 {
-    PRECONDITION(not this->fields.empty(), "Cannot generate a row if there are no fields!");
-    const auto generateField = Overloaded{
-        [this, &ostream](GeneratorFields::BaseStoppableGeneratorField& field)
-        {
-            const bool fieldAlreadyStopped = field.stop;
-            field.generate(ostream, this->randEng);
-            if (field.stop && !fieldAlreadyStopped)
+    if (fromFile.empty())
+    {
+        PRECONDITION(not this->fields.empty(), "Cannot generate a row if there are no fields!");
+        const auto generateField = Overloaded{
+            [this, &ostream](GeneratorFields::BaseStoppableGeneratorField& field)
             {
-                this->numStoppedFields++;
-            }
-        },
-        [this, &ostream](GeneratorFields::BaseGeneratorField& field) { field.generate(ostream, this->randEng); }};
+                const bool fieldAlreadyStopped = field.stop;
+                field.generate(ostream, this->randEng);
+                if (field.stop && !fieldAlreadyStopped)
+                {
+                    this->numStoppedFields++;
+                }
+            },
+            [this, &ostream](GeneratorFields::BaseGeneratorField& field) { field.generate(ostream, this->randEng); }};
 
-    std::visit(generateField, *this->fields.front());
-    for (auto& field : this->fields | std::views::drop(1))
+        std::visit(generateField, *this->fields.front());
+        for (auto& field : this->fields | std::views::drop(1))
+        {
+            ostream << Generator::fieldDelimiter;
+            std::visit(generateField, *field);
+        }
+        ostream << Generator::tupleDelimiter;
+    }
+    else
     {
-        ostream << Generator::fieldDelimiter;
-        std::visit(generateField, *field);
+        if (not fileStream.is_open())
+        {
+            fileStream.open(fromFile);
+            /// Next line always fails, so as a workaround instead we use the if statement below.
+            ///INVARIANT(not fileStream.is_open(), "Failed to open file");
```

Comment:
```suggestion
            NVARIANT(fileStream.is_open(), "Failed to open file");
```

This code assumes that at this moment the fileStream is open